src/Authentication/Authentication/Cmdlets/InvokeGraphRequest.cs

```diff
@@ -0,0 +1,965 @@
+            {
+                // save the session back to the PS environment if requested
+                var vi = SessionState.PSVariable;
+                vi.Set(SessionVariable, GraphRequestSession);
+            }
+
+            if (Authentication == GraphRequestAuthenticationType.UserProvidedToken && !string.IsNullOrWhiteSpace(Token))
+            {
+                GraphRequestSession.Token = Token;
+                GraphRequestSession.AuthenticationType = Authentication;
+            }
+            
+            //
+            // Handle Custom User Agents
+            //
+            GraphRequestSession.UserAgent = UserAgent ?? _graphRequestUserAgent.UserAgent;
+
+            // Store the other supplied headers
+            if (Headers != null)
+            {
+                foreach (var key in Headers.Keys)
+                {
+                    var value = Headers[key];
+
+                    // null is not valid value for header.
+                    // We silently ignore header if value is null.
+                    if (!(value is null))
+                    {
+                        // add the header value (or overwrite it if already present)
+                        GraphRequestSession.Headers[key] = value;
+                    }
+                }
+            }
+        }
+        /// <summary>
+        /// Validate the Request Uri must have the same Host as GraphHttpClient BaseAddress.
+        /// </summary>
+        /// <param name="httpClient"></param>
+        private void ValidateRequestUri(HttpClient httpClient)
+        {
+            if (Uri == null)
+            {
+                var error = GetValidationError($"Must specify {nameof(Uri)}", "InvokeGraphRequestInvalidHost",
+                    nameof(Uri));
+                ThrowTerminatingError(error);
+            }
+            // Ensure that the Passed in Uri has the same Host as the HttpClient. 
+            if (Uri.IsAbsoluteUri && httpClient.BaseAddress.Host != Uri.Host)
+            {
+                var error = GetValidationError($"Invalid Host {Uri.Host}", "InvokeGraphRequestInvalidHost",
+                    nameof(Uri));
+                ThrowTerminatingError(error);
+            }
+        }
+        /// <summary>
+        /// Validate Passed In Parameters
+        /// </summary>
+        private void ValidateParameters()
+        {
+            if (GraphRequestSession != null && SessionVariable != null)
+            {
+                var error = GetValidationError(
+                    "The cmdlet cannot run because the following conflicting parameters are specified: Session and SessionVariable. Specify either Session or SessionVariable, then retry.",
+                    "WebCmdletSessionConflictException");
+                ThrowTerminatingError(error);
+            }
+
+            // When PATCH or POST is specified, ensure a body is present
+            if (Method == GraphRequestMethod.PATCH || Method == GraphRequestMethod.POST && Body == null)
```

Comment:
This logic doesn't do what is mentioned in the comment above. `&&` comes higher in the order of precedence against `||`. If Method is `PATCH`, the other conditions will not be checked.

```suggestion
            if ((Method == GraphRequestMethod.PATCH || Method == GraphRequestMethod.POST) && Body == null)
```